File: wordcount.pl
#!/usr/bin/env perl

my %counter;

my $line = <STDIN>;
while($line ne "") {
    my @words = split(/\s+/, $line);
    for my $word (@words) {
        $counter{$word} ++;
    }
    $line = <STDIN>;
}

foreach my $word (sort { -$counter{$a} <=> -$counter{$b} or $a cmp $b } keys %counter) {
    printf "%s\t%s\n", $word, $counter{$word};
}

File: wordcount.php
<?php

function countWord($word,$array){

    if (!array_key_exists($word,$array)){
        $array[$word] = 1;
        return;
    }
    $array[$word]++;
}


$stdin = fopen('php://stdin', 'r');
//$stdin = fopen('de', 'r');



$array = array();
while (false !== ($line = fgets($stdin))) {
    $words = explode(" ", $line);

    foreach($words as $word){
        if (!array_key_exists($word,$array)){
            $array[$word] = 1;
            continue;
        }
        $array[$word]++;
    }

}
ksort($array);
arsort($array);
foreach($array as $key => $value){
    echo $key."\t".$value."\n";
}